app/helpers/template_helpers.py

```diff
@@ -14,291 +14,343 @@
+        if self.context_options.title_logo:
+            context["titleLogo"] = f"{self.context_options.title_logo}"
+        if self.context_options.title_logo_alt:
+            context["titleLogoAlt"] = f"{self.context_options.title_logo_alt}"
+        if self.context_options.header_logo:
+            context["customHeaderLogo"] = self.context_options.header_logo
+        if self.context_options.mobile_logo:
+            context["mobileLogo"] = self.context_options.mobile_logo
+
+        return context
+
+    @property
+    def footer_context(self):
+        context = {
+            "lang": self.context_options.lang,
+            "crest": self.context_options.crest,
+            "newTabWarning": lazy_gettext("The following links open in a new tab"),
+            "copyrightDeclaration": {
+                "copyright": self.context_options.copyright_declaration,
+                "text": self.context_options.copyright_text,
             },
-        },
-    }
-    return context.get(theme)
+        }
 
+        if self._footer_warning:
+            context["footerWarning"] = self._footer_warning
+
+        if self.context_options.footer_links:
+            for footer_link in self.context_options.footer_links:
+                footer_link["url"] = self._base_url + footer_link["url"]
+            context["rows"] = [{"itemsList": self.context_options.footer_links}]
+
+        if self.context_options.footer_legal_links:
+            for footer_legal_link in self.context_options.footer_legal_links:
+                url = footer_legal_link["url"]
+                footer_legal_link["url"] = self._base_url + url
+            context["legal"] = [{"itemsList": self.context_options.footer_legal_links}]
+
+        if (
+            self.context_options.powered_by_logo
+            or self.context_options.powered_by_logo_alt
+        ):
+            context["poweredBy"] = {
+                "logo": self.context_options.powered_by_logo,
+                "alt": self.context_options.powered_by_logo_alt,
+            }
+
+        return context
+
+    @cached_property
+    def _footer_warning(self) -> Union[str, None]:
+        if request.blueprint == "post_submission":
+            return lazy_gettext(
+                "Make sure you <a href='{sign_out_url}'>leave this page</a> or close your browser if using a shared device"
+            ).format(sign_out_url=self._sign_out_url)
+
+
+@dataclass
+class CensusContextOptions(ContextOptions):
+    title_logo: str = "census-logo-en"
+    title_logo_alt: str = lazy_gettext("Census 2021")
+    account_service_url = f"{CENSUS_EN_BASE_URL}/en/start"
+    design_system_theme = "census"
+    footer_links: Iterable[Mapping] = field(
+        default_factory=lambda: [
+            Link(
+                lazy_gettext("Help"),
+                "help/help-with-the-questions/online-questions-help/",
+            ).__dict__,
+            Link(lazy_gettext("Contact us"), "contact-us/").__dict__,
```

Comment:
What's your reason for not using the dunder method?

Reply:
Developer preference I guess, but I don't think it's very readable to use dunders directly. I would say it is more of an internal representation which should have a wrapper and not invoked directly outside the object. Plus it is hard without checking code if there is our own implementation to support `dict()` on the object. The other objects we have, all use a wrapper method so thought it would also make that consistent. If we are keeping `__dict__` then there is one reference to `asdict()` which should be updated to use the dunder for consistency.